src/matchers/toBeAccessible/index.ts

```diff
@@ -1,44 +1,41 @@
 import test from '@playwright/test'
 import type { MatcherState } from '@playwright/test/types/expect-types'
-import type { Result, RunOptions } from 'axe-core'
+import type { AxeResults, Result } from 'axe-core'
+import type { MatcherOptions } from '../../types'
+import type { Handle } from '../../utils/locator'
 import createHTMLReport from 'axe-reporter-html'
 import merge from 'merge-deep'
 import { attach } from '../../utils/attachments'
-import { injectAxe, runAxe } from '../../utils/axe'
-import { Handle, resolveLocator } from '../../utils/matcher'
-import { poll } from '../../utils/poll'
+import { waitForAxeResults } from '../../waitForAxeResults'
 
 const summarize = (violations: Result[]) =>
   violations
     .map((violation) => `${violation.id}(${violation.nodes.length})`)
     .join(', ')
 
-interface MatcherOptions extends RunOptions {
-  timeout?: number
-  filename?: string
+async function getResults(obj: Handle | AxeResults, options: MatcherOptions) {
+  if ((obj as AxeResults).violations) {
+    const results = obj as AxeResults
+    return {
+      results,
+      ok: !results.violations.length,
+    }
+  } else {
+    return waitForAxeResults(obj as Handle, options)
+  }
 }
 
 export async function toBeAccessible(
   this: MatcherState,
-  handle: Handle,
-  { timeout, ...options }: MatcherOptions = {}
+  obj: Handle | AxeResults,
+  options: MatcherOptions = {}
 ) {
   try {
-    const locator = resolveLocator(handle)
-    await injectAxe(locator)
+    const { results, ok } = await getResults(obj, options)
 
     const info = test.info()
     const opts = merge(info.project.use.axeOptions, options)
```

Comment:
This merge needs to happen before calling `getResults` and the `opts` object needs to be passed to `getResults`.  This allows for defining config at the project level.

Reply:
Oh, looks like `getResults` is doing that already.  Interesting

Reply:
Yeah, there's some repetition, but I'm not sure how to avoid it. 

Do you want me to try to refactor this in a way that avoids the repetition? Or do you want some time to think about it?